core/src/test/java/org/elasticsearch/index/get/GetFieldTests.java

```diff
@@ -0,0 +1,192 @@
+            parsedGetField = GetField.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_ARRAY, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+        assertEquals(expectedGetField, parsedGetField);
+        BytesReference finalBytes = toXContent(parsedGetField, xContentType, true);
+        assertSameOutput(originalBytes, finalBytes, xContentType);
+    }
+
+    private static GetField copyGetField(GetField getField) {
+        return new GetField(getField.getName(), getField.getValues());
+    }
+
+    private static GetField mutateGetField(GetField getField) {
+        List<Supplier<GetField>> mutations = new ArrayList<>();
+        mutations.add(() -> new GetField(randomUnicodeOfCodepointLength(15), getField.getValues()));
+        mutations.add(() -> new GetField(getField.getName(), randomGetField(XContentType.JSON).v1().getValues()));
+        return randomFrom(mutations).get();
+    }
+
+    public static Tuple<GetField, GetField> randomGetField(XContentType xContentType) {
+        if (randomBoolean()) {
+            String fieldName = randomFrom(ParentFieldMapper.NAME, RoutingFieldMapper.NAME, UidFieldMapper.NAME);
+            GetField getField = new GetField(fieldName, Collections.singletonList(randomAsciiOfLengthBetween(3, 10)));
+            return Tuple.tuple(getField, getField);
+        }
+        String fieldName = randomAsciiOfLengthBetween(3, 10);
+        Tuple<List<Object>, List<Object>> tuple = randomStoredFieldValues(xContentType);
+        GetField input = new GetField(fieldName, tuple.v1());
+        GetField expected = new GetField(fieldName, tuple.v2());
+        return Tuple.tuple(input, expected);
+    }
+
+    //TODO move this to some utility class
+    /**
+     * Asserts that the provided {@link BytesReference}s hold the same content. The comparison is done between the map
+     * representation of the provided objects.
+     */
+    public static void assertSameOutput(BytesReference expected, BytesReference actual, XContentType xContentType) throws IOException {
+        //we tried comparing byte per byte, but that didn't fly for a couple of reasons:
+        //1) whenever anything goes through a map, ordering is not preserved, which is perfectly ok
+        //2) Jackson SMILE parser parses floats as double, which then get printed out as double (with double precision)
+        try (XContentParser parser = xContentType.xContent().createParser(actual)) {
+            Map<String, Object> finalMap = parser.map();
+            try (XContentParser parser2 = xContentType.xContent().createParser(expected)) {
+                Map<String, Object> originalMap = parser2.map();
+                assertEquals(originalMap, finalMap);
+            }
+        }
+    }
+
+    //TODO move this to some utility class? It should be useful in other tests too, whenever we parse stored fields
```

Comment:
there's a couple of TODOs like this, I am not even sure it is worth doing, the main issue being I don't where to put the class and what to call it :)